spec/controllers/posts_controller_spec.rb

```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
```

Comment:
Add some tests to this controller :)

Reply:
Some tips 💡 

What happens when:
- User is logged in for each action in controller
- USer is not logged in for each action in controller
- Think positive and negative scenarios

Here is an example:
```ruby
describe 'GET show' do 

    context 'when user is logged in' do 
      
      before do 
        sign_in a_user
        get :show, params: { username: a_user.username, id: a_post.id }
      end

      

      it 'renders :show template' do
        expect(response).to render_template(:show)
      end
    end

    context 'when user is NOT logged in' do 

      before do 
        get :show, params: { username: a_user.username, id: a_post.id }
      end

      it 'redirects to profile_posts_path' do 
        expect(response).to redirect_to(profile_path)
      end
    end
  end
```
This is, of course, an out of context example, but hopefully, it can give you an idea of how you can do your tests.

[OPTIONAL]